homeassistant/components/fastdotcom/services.py

```diff
@@ -0,0 +1,49 @@
+"""Service for Fast.com integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import issue_registry as ir
+
+from .const import DOMAIN, SERVICE_NAME
+from .coordinator import FastdotcomDataUpdateCoordindator
+
+
+async def async_setup_services(hass: HomeAssistant) -> None:
+    """Set up the services for the Fast.com integration."""
+
+    async def collect_coordinator() -> FastdotcomDataUpdateCoordindator:
```

Comment:
If we register the service after startup this whole thing would not be needed

Reply:
We don't register it anymore after startup? Basically all should be in `services.py` now.

Reply:
It goes hand in hand with the comment above so let's see what reply we get there first as I'm curious to know why we would load services from `async_setup()` and not as part of creating it from the entry.